File: php7/src/DefaultItem.php
<?php


namespace App;


class DefaultItem extends GildedRose
{
    private $item;

    public function __construct($item) {
        $this->item = $item;
    }

    public function updateQuality(){
        $this->item->quality -= 1;

        if ($this->item->sell_in <= 0) {
            $this->item->quality -= 1;
        }

        if($this->item->quality >= 50){
            $this->item->quality = 50;
        }

        $this->item->sell_in -= 1;

    }
}

File: Java/src/main/java/com/gildedrose/domain/DefaultItem.java
package com.gildedrose.domain;

import com.gildedrose.Item;

public class DefaultItem extends InventoryItem {

    public DefaultItem(Item item) {
        setName(item.name);
        setSellIn(item.sellIn);
        setQuality(item.quality);
    }

    @Override
    public int qualityDecreaseAmount() {
        return 1;
    }

    @Override
    public int handleQualityAfterSellIn() {
        return decreaseQualityAboveZero();
    }
}
